modules/pubsub/broker.go

```diff
@@ -0,0 +1,101 @@
+package pubsub
+
+import (
+	"sync"
+
+	"github.com/spaceuptech/space-cloud/model"
+	"github.com/spaceuptech/space-cloud/utils"
+)
+
+natsWildcard string = ">"
+
+type subscription interface {
+	Unsubscribe() error
+}
+
+type pubsubBroker interface {
+	Publish(subject string, msg *model.PubsubMsg) error
+	QueueSubscribe(subject, group string, ch chan *model.PubsubMsg) (*subscription, error)
+}
+
+type pubsubSubscription {
+	subs *subscription
+	ch   chan *model.PubsubMsg
+}
+
+func (m *Module) Publish(subject string, data *model.PubsubMsg) error {
+	return m.connection.Publish(subject, data)
+}
+
+func (m *Module) Subscribe(clientId, subject string, cb model.PubsubCallback) (error) {
+	return m.QueueSubscribe(clientId, subject, clientId, cb)
+}
+
+func (m *Module) QueueSubscribe(clientId, subject, group string, cb model.PubsubCallback) (error) {
+	ch := make(chan *model.PubsubMsg, 10)
+	subject = utils.SingleLeadingTrailing(subject, "/")
+	subs, err := m.connection.QueueSubscribe(subject + natsWildcard, group, ch)
+	if err != nil {
+		return err
+	}
+	go func() {
+		for msg := range ch {
+        	cb(msg.Subject, msg)
```

Comment:
Use the subject from the parameter

Reply:
If user subscibes to "/a/"(the subject in the parameter), and some one publishes to "/a/b/", we send the subject as "/a/b/"(the subject in the msg), right?

Reply:
You are correct. My bad